### DISCUSSION TITLE
Unable to retrieve message using ChannelID and MessageID

### DISCUSSION THREAD

**Discussion by Technerder** [bwmarrin/discordgo#1326] [Q&A]
Over the last few hours I've been trying to work on a discord bot that needs to be able to retrieve message contents (including reactions) based off of a messages ChannelID and MessageID, but can't seem to be able to do that. The bot program is only able to retrieve messages that were sent after it was started and running. Attempting to retrieve older messages (i.e. before the bot was started, or many hours ago) using the `session.State.Message` function results in a `state cache not found` error being returned by the function. I've attached a minimal example that manages to reproduce the problem I'm facing and would love to know what exactly it is that I'm doing wrong.

Any and all help is much appreciated!

Bot Invite URL being used: https://discord.com/api/oauth2/authorize?client_id=BOT_ID&permissions=68608&scope=applications.commands%20bot

Reproducible code:
```go
package main

import (
	"github.com/bwmarrin/discordgo"
	"log"
	"os"
	"os/signal"
	"syscall"
)

var (
	BotToken         = ""
	MessageChannelID = ""
	MessageID        = ""
)

func main() {
	bot, err := discordgo.New("Bot " + BotToken)
	if err != nil {
		log.Printf("An error occurred while trying to create bot: %v\n", err)
		return
	}
	defer bot.Close()

	bot.AddHandler(func(session *discordgo.Session, ready *discordgo.Ready) {
		log.Println("Bot connected... ")

		message, err := session.State.Message(MessageChannelID, MessageID)
		if err != nil {
			log.Printf("An error occurred while trying to retrieve message: %v\n", err)
			return
		}
		log.Printf("Retrieved Message Content: %s\n", message.Content)
	})

	bot.Identify.Intents = discordgo.IntentsAll
	bot.State.MaxMessageCount = 1000

	err = bot.Open()
	if err != nil {
		log.Printf("An error occurred while trying to open bot connection: %v\n", err)
		return
	}

	log.Println("Press CTRL-C to exit...")
	sc := make(chan os.Signal, 1)
	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
	<-sc
}

```

**Comment by FedorLap2006** [comment]
Sorry for a late response.

"state not found" or `discordgo.ErrStateNotFound` is an error which is thrown when the object is not present in the state (it's mostly empty right after the startup, which is indicated by the `Ready` event). To prevent such problems from happening, it's a good idea to fallback (if state fails) to fetching the object (message) from the API directly (`s.ChannelMessage`).
